components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/internal/handler/CredentialEventHookHandler.java

```diff
@@ -167,13 +167,7 @@ private boolean isSupportedEvent(String eventName) {
 
     public boolean isCredentialUpdateFlow(String eventName) {
 
-        if (IdentityEventConstants.Event.POST_ADD_NEW_PASSWORD.equals(eventName)) {
-            Flow flow = IdentityContext.getThreadLocalIdentityContext().getFlow();
-            Flow.Name flowName = (flow != null) ? flow.getName() : null;
-
-            return !Flow.Name.USER_REGISTRATION_INVITE_WITH_PASSWORD.equals(flowName);
-        }
-
-        return IdentityEventConstants.Event.POST_UPDATE_CREDENTIAL_BY_SCIM.equals(eventName);
+        return IdentityEventConstants.Event.POST_ADD_NEW_PASSWORD.equals(eventName) ||
+                IdentityEventConstants.Event.POST_UPDATE_CREDENTIAL_BY_SCIM.equals(eventName);
```

Comment:
There's a chance a credential update to be triggered in cases like provisioning a user with internal generated password. Can happen at JIT provisioning. Such cases are not actually password update events. So, IMO you have to check the flow here. And this should be treated as a credential update flow if flow really is admin invite with password

Reply:
FIxed d2abc9d3562a4cd76a62344e5cc2d88e893cc71c